cmd/plugin/cmd/init.go

```diff
@@ -67,65 +93,417 @@ var initCmd = &cobra.Command{
+			if cond.Type == clusterv1.ReadyCondition && cond.Status == corev1.ConditionTrue {
+				log.Info("Provider is ready", "Type", genericProvider.GetType(), "Name", genericProvider.GetName(), "Namespace", genericProvider.GetNamespace())
+
+				return true, nil
+			}
+		}
+
+		return false, nil
+	}); err != nil {
+		log.Error(err, "Provider is not ready", "Type", genericProvider.GetType(), "Name", genericProvider.GetName(), "Namespace", genericProvider.GetNamespace())
+	}
+}
+
+func ensureCertManager(ctx context.Context, opts *initOptions) error {
+	configClient, err := configclient.New(ctx, "")
+	if err != nil {
+		return fmt.Errorf("cannot create config client: %w", err)
+	}
+
+	clusterKubeconfig := cluster.Kubeconfig{
+		Path:    opts.kubeconfig,
+		Context: opts.kubeconfigContext,
+	}
+
+	clusterClient := cluster.New(clusterKubeconfig, configClient)
+
+	// Before installing the operator, ensure the cert-manager Webhook is in place.
+	certManager := clusterClient.CertManager()
+	if err := certManager.EnsureInstalled(ctx); err != nil {
+		return fmt.Errorf("cannot install cert-manager Webhook: %w", err)
+	}
+
+	return nil
+}
+
+// deployCAPIOperator deploys the CAPI operator on the management cluster.
+func deployCAPIOperator(ctx context.Context, opts *initOptions) error {
+	configClient, err := configclient.New(ctx, "")
+	if err != nil {
+		return fmt.Errorf("cannot create config client: %w", err)
+	}
+
+	capiOperatorManifestsURL := fmt.Sprintf(capiOperatorManifestsURLTemplate, opts.operatorVersion)
+
+	providerConfig := configclient.NewProvider(capiOperatorProviderName, capiOperatorManifestsURL, clusterctlv1.ProviderTypeUnknown)
+
+	repo, err := util.RepositoryFactory(ctx, providerConfig, configClient.Variables())
+	if err != nil {
+		return fmt.Errorf("cannot create repository: %w", err)
+	}
+
+	componentsFile, err := repo.GetFile(ctx, opts.operatorVersion, repo.ComponentsPath())
+	if err != nil {
+		return fmt.Errorf("cannot get components file: %w", err)
+	}
+
+	options := repository.ComponentsOptions{
+		TargetNamespace:     opts.targetNamespace,
+		SkipTemplateProcess: false,
+		Version:             opts.operatorVersion,
+	}
+
+	components, err := repository.NewComponents(repository.ComponentsInput{
+		Provider:     providerConfig,
+		ConfigClient: configClient,
+		Processor:    yamlprocessor.NewSimpleProcessor(),
+		RawYaml:      componentsFile,
+		Options:      options,
+	})
+	if err != nil {
+		return fmt.Errorf("cannot generate CAPI operator components: %w", err)
+	}
+
+	clusterKubeconfig := cluster.Kubeconfig{
+		Path:    opts.kubeconfig,
+		Context: opts.kubeconfigContext,
+	}
+
+	clusterClient := cluster.New(clusterKubeconfig, configClient)
+
+	if err := clusterClient.ProviderComponents().Create(ctx, components.Objs()); err != nil {
+		return fmt.Errorf("cannot create CAPI operator components: %w", err)
+	}
+
+	return nil
+}
+
+func newGenericProvider(providerType clusterctlv1.ProviderType) operatorv1.GenericProvider {
```

Comment:
We have a similar (but opposite of that) method in utils. Does it make sense to move it there? Also maybe we can leverage `GetType` on the provider wrapper to only store a list of providers

Reply:
Moved this method to utils.go